### PR TITLE
feat: add interactive cli for rollapp init

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: cmd/config/init/flags.go [hunk 1]
```diff
@@ -19,6 +16,7 @@ func addFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP(FlagNames.HubID, "", StagingHubID, fmt.Sprintf("The ID of the Dymension hub. %s", getAvailableHubsMessage()))
 	cmd.Flags().StringP(FlagNames.RollappBinary, "", "", "The rollapp binary. Should be passed only if you built a custom rollapp")
 	cmd.Flags().StringP(FlagNames.TokenSupply, "", defaultTokenSupply, "The total token supply of the RollApp")
+	cmd.Flags().BoolP(FlagNames.Interactive, "", false, "Run roller in interactive mode")
```
Comment (ItayLevyOfficial) [inline cmd/config/init/flags.go:19]: should default to true

File: cmd/config/init/interactive.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	promptNetwork := promptui.Select{
+		Label: "Network",
+		Items: []string{"local", "internal-devnet"},
+	}
+	_, mode, _ := promptNetwork.Run()
+	//TODO: add validation
+	config.HubData = Hubs[mode]
+
+	// promptPrefix := promptui.Prompt{
```
Comment (ItayLevyOfficial) [inline cmd/config/init/interactive.go:53]: remove comments

File: cmd/config/init/init.go [hunk 3]
```diff
@@ -14,23 +15,28 @@ func InitCmd() *cobra.Command {
...
+			//TODO: parse the config here instead of GetInitConfig in Run command
+			// cmd.SetContextValue("mydata", data)
+
 			return nil
 		},
 		Run: func(cmd *cobra.Command, args []string) {
 			initConfig, err := GetInitConfig(cmd, args)
 			utils.PrettifyErrorIfExists(err)
+
+			err = initConfig.Validate()
```
Comment (ItayLevyOfficial) [inline cmd/config/init/init.go:36]: don't we want to yell on invalid values as the user pass them in the interactive flow? so he will be able to fix them?

File: cmd/config/init/flags.go (hunk 1 above)
Comment (mtsitrin) [reply cmd/config/init/flags.go:19]: @yishay-dym @shaolin-flow what the default for the docs?

File: cmd/config/init/init.go (hunk 3 above)
Comment (mtsitrin) [reply cmd/config/init/init.go:36]: did it here for now, I think it's fine for the MVP

**Comment by omritoptix** [comment]
@mtsitrin please wait to get the relevant labels/options and order for the interactive prompts from @yishay-dym before merging this PR.

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM

**Comment by omritoptix** [comment]
@mtsitrin this is the labels:


1. Select your network:
2. Enter your RollApp ID (format: rollapp-name_EIP155-revision. e.g wagmi_420-1):
3. Specify your RollApp denom (e.g BTC, PEPE, DYM):
4. How many tokens do you wish to mint for Genesis? (default: 1,000,000,000):
5. Choose your data layer:
6. Set your runtime binary. for EVM RollApp hit ‘Enter’. For a custom app-specific RollApp, input it’s binary path:

**Comment by mtsitrin** [comment]
what about
```
	promptType := promptui.Select{
		Label: "Rollapp runtime type",
		Items: []string{"evm", "sdk"},
	}
	_, _, _ = promptType.Run()
	fmt.Println("Only EVM supported for now")
	```

**Comment by omritoptix** [comment]
I'm checking if providing a custom binary will work. if it won't we can go with what you suggest for now.

**Comment by omritoptix** [comment]
@mtsitrin lets go with your suggestion for now. if it works worst case we'll change later.

**Comment by mtsitrin** [comment]
It's just a stub currently anyway, until we'll have SDK support
I'll remove it for now

**Review by omritoptix — APPROVED** [review]

**Pull request merged**